AnnService/src/Core/VectorIndex.cpp

```diff
@@ -269,9 +276,13 @@ VectorIndex::BuildIndex(std::shared_ptr<VectorSet> p_vectorSet,
 
 
 ErrorCode
-VectorIndex::SearchIndex(const void* p_vector, int p_neighborCount, bool p_withMeta, BasicResult* p_results) const {
-    QueryResult res(p_vector, p_neighborCount, p_withMeta, p_results);
-    SearchIndex(res);
+VectorIndex::SearchIndex(const void* p_vector, int p_vectorCount, int p_neighborCount, bool p_withMeta, BasicResult* p_results) const {
+    size_t vectorSize = GetValueTypeSize(GetVectorValueType()) * GetFeatureDim();
+#pragma omp parallel for schedule(dynamic,10)
+    for (int i = 0; i < p_vectorCount; i++) {
```

Comment:
searchindex need follow the policy to leverage only one thread, even for batch search(though fresh ANN index didn't support batch search), as we use request thread pool to control processor usage, every request thread consume several core is not our expectation

Reply:
This interface is only used for parallel search in the python and java wrappers to speedup the offline search.